modules/avformat/avformat.c

```diff
@@ -203,7 +205,11 @@ int avformat_shared_alloc(struct shared **shp, const char *dev)
 	if (err)
 		goto out;
 
-	ret = avformat_open_input(&st->ic, dev, NULL, NULL);
+	input_format = av_find_input_format(dev);
+	if (!input_format)
+		input_format = NULL;
```

Comment:
I think these 2 lines (209-210) are redundant, and can be removed.

in this case, input_format is already NULL